desc/backend.py

```diff
@@ -269,18 +273,25 @@ def bodyfun(state):
                 xk1, fk1 = backtrack(xk1, fk1, d)
                 return xk1, fk1, k1 + 1
 
-            state = guess, res(guess), 0
+            state = guess, res(guess), 0.0
```

Comment:
is making this a float also required for it to be differentiable?

Reply:
So the changes I made prevent the problem in 2 different way. We don't need both of them implemented but kept them.
1. make `niter` float such that it wouldn't cause the issue
2. add `full_output` flag and call `map_coordinates` with `full_output=False`
Actually, 1 is enough to solve the problem fully. I implemented 2 first, idk maybe it helps with something else in future?!